codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/GenerationContext.java

```diff
@@ -161,4 +164,25 @@ public List<RubyRuntimePlugin> getRuntimePlugins() {
                 .flatMap(List::stream)
                 .collect(Collectors.toList());
     }
+
+    /**
+     * @return list of all AuthSchemes from all integrations and the application transport.
+     */
+    public List<AuthScheme> getAuthSchemes() {
+        List<AuthScheme> authSchemes = applicationTransport().defaultAuthSchemes();
+        authSchemes.add(AnonymousAuthSchemeFactory.build());
+        integrations().forEach((i) -> {
+            i.getAdditionalAuthSchemes(this).forEach((s) -> authSchemes.add(s));
+        });
+
+        ServiceIndex.of(model()).getAuthSchemes(service()).forEach((shapeId, trait) -> {
+            AuthScheme authScheme = authSchemes.stream()
+                    .filter(s -> s.getShapeId().equals(shapeId))
+                    .findFirst()
+                    .orElseThrow(() -> new IllegalStateException("No auth scheme found for " + shapeId));
+            authScheme.extractSignerProperties(trait);
+            authScheme.extractIdentityProperties(trait);
+        });
+        return authSchemes;
```

Comment:
Ditto - this shouldn't be modifiable.

Reply:
getRuntimePlugins is modifiable. Why can't this?

Reply:
`getRuntimePlugins` shouldn't be modifiable either.  It looks like its using `.collect(Collectors.toList());` which I had assumed created an unmodifiable list, but it does not.  

Can you change this to `.collect(Collectors.toUnmodifiableList());`